sympy/core/numbers.py

```diff
@@ -778,14 +783,16 @@ def __new__(cls, num, dps=None, precision=None):
             elif not _literal_float(num):
                 raise ValueError('string-float not recognized: %s' % _num)
         elif isinstance(num, float) and num == 0:
-            num = '0'
+            return S.Zero  # there is not Float(0)
```

Comment:
I am not sure about removing `Float(0)`.

Reply:
see https://github.com/sympy/sympy/pull/25614#issuecomment-1703756842

Reply:
Instead of removing it maybe we could all `Float(0) == 0` to be the only exception. Maybe that was @jksuom intent from the start.

Reply:
What I don't like is that we lose the distinction between exact and approximate.

Reply:
Then we need to make `Float(0)*1 -> Float(0)`. I can undo the `Float(0) == 0` and leave make `Float(0)*1 = Float(0)` for another PR. It, too, is not a trivial change because of the code that depends on this behavior.